starboard/nplb/posix_compliance/posix_epoll_wait_test.cc

```diff
@@ -0,0 +1,286 @@
+// Copyright 2025 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* The following error scenarios are untested for epoll_wait():
+1. EFAULT (events pointer is bad) is difficult to safely and reliably test.
+2. EINTR (interrupted by signal) is difficult to test.
+3. Negative timeout is an infinite timeout, which is difficult to test.
+*/
+
+#include <fcntl.h>
+#include <sys/epoll.h>
+#include <unistd.h>
+
+#include "starboard/nplb/posix_compliance/posix_epoll_test_helper.cc"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace starboard {
+namespace nplb {
+namespace {
+
+class PosixEpollWaitTests : public PosixEpollTest {
+ protected:
+  int epfd_ = -1;
+  int pipe_fds_[2] = {-1, -1};  // pipe_fds_[0] for read, pipe_fds_[1] for write
+  struct epoll_event events_[kMaxEvents];
+
+  void SetUp() override {
+    epfd_ = epoll_create(1);
+    ASSERT_NE(epfd_, -1) << "Failed to create epoll instance in SetUp: "
+                         << strerror(errno);
+    // Use non-blocking pipes to prevent writes from blocking indefinitely if
+    // pipe is full
+    ASSERT_TRUE(CreatePipe(pipe_fds_, O_NONBLOCK))
+        << "Failed to create pipe in SetUp: " << strerror(errno);
+  }
+
+  void TearDown() override {
+    if (epfd_ >= 0) {
+      close(epfd_);
+      epfd_ = -1;
+    }
+    if (pipe_fds_[0] >= 0) {
+      close(pipe_fds_[0]);
+      pipe_fds_[0] = -1;
+    }
+    if (pipe_fds_[1] >= 0) {
+      close(pipe_fds_[1]);
+      pipe_fds_[1] = -1;
+    }
+  }
+
+  // Helper to add to epoll
+  void AddToEpoll(int fd_to_add, uint32_t ev_mask) {
+    struct epoll_event event = {};
+    event.events = ev_mask;
+    event.data.fd = fd_to_add;
+    ASSERT_EQ(epoll_ctl(epfd_, EPOLL_CTL_ADD, fd_to_add, &event), 0)
+        << "Failed to add fd " << fd_to_add << " to epoll with events "
+        << ev_mask << ": " << strerror(errno);
+  }
+
+  // Helper to remove from epoll
+  void RemoveFromEpoll(int fd_to_remove) {
+    struct epoll_event event {};  // Dummy event, required for older kernels
+    if (epfd_ >= 0 && fd_to_remove >= 0) {
+      epoll_ctl(epfd_, EPOLL_CTL_DEL, fd_to_remove, &event);
+    }
+  }
+};
+
+TEST_F(PosixEpollWaitTests, EventReportedSuccessfully) {
+  AddToEpoll(pipe_fds_[0], EPOLLIN);  // Watch read end for input
+
+  char buffer = 'x';
+  ASSERT_EQ(write(pipe_fds_[1], &buffer, 1), 1)
```

Comment:
micro-nit: Ideally this "1"s would be labelled, e.g.
```
  const size_t buffer_length = sizeof(buffer);
  ASSERT_EQ(write(pipe_fds_[1], &buffer, buffer_length), buffer_length)
      << ...
```
Otherwise is unclear whether they are the same (or e.g.
the second one is a different number).